### PR TITLE
[MOS][CodeGen] Implement structured bindings in for-loops. NFC.

### PR DESCRIPTION
**Pull request by gandhi56**

### CODE REVIEW

File: llvm/lib/Target/MOS/MOSCallLowering.cpp [hunk 1]
```diff
@@ -296,11 +296,10 @@ bool MOSCallLowering::lowerReturn(MachineIRBuilder &MIRBuilder,
     // Copy flags from the instruction definition over to the return value
     // description for TableGen compatibility layer.
     SmallVector<ArgInfo> Args;
-    // TODO: C++17 structured bindings
-    for (const auto &I : zip(VRegs, ValueVTs, ValueLLTs)) {
-      Args.emplace_back(std::get<0>(I), std::get<1>(I).getTypeForEVT(Ctx), 0);
+	for (const auto &[VReg, ValueVT, ValueLLT] : zip(VRegs, ValueVTs, ValueLLTs)) {
```
Comment (mysterymath) [inline llvm/lib/Target/MOS/MOSCallLowering.cpp:299]: It looks like the for loop is too indented here.

File: llvm/lib/Target/MOS/MOSLegalizerInfo.cpp [hunk 2]
```diff
@@ -1118,10 +1118,9 @@ bool MOSLegalizerInfo::legalizeICmp(LegalizerHelper &Helper,
     auto RHSUnmergeDefs = unmergeDefsSplitHigh(RHSUnmerge);
     assert(LHSUnmerge->getNumOperands() == RHSUnmerge->getNumOperands());
     CIn = Builder.buildConstant(S1, 1).getReg(0);
-    // TODO: C++17 structured bindings
-    for (const auto &I : zip(LHSUnmergeDefs.Lows, RHSUnmergeDefs.Lows)) {
+	for (const auto &[LHS, RHS] : zip(LHSUnmergeDefs.Lows, RHSUnmergeDefs.Lows)) {
```
Comment (mysterymath) [inline llvm/lib/Target/MOS/MOSLegalizerInfo.cpp:1121]: Too indented here too.

File: llvm/lib/Target/MOS/MOSLowerSelect.cpp [hunk 3]
```diff
@@ -314,12 +314,8 @@ MOSLowerSelect::lowerSelect(MachineInstr &MI) {
...
     //  ...
     Builder.setInsertPt(*SinkMBB, SinkMBB->begin());
-    // TODO: C++17 structured bindings
-    for (const auto &I : zip(Dsts, TrueValues, FalseValues)) {
-      Register Dst = std::get<0>(I);
-      Register TrueValue = std::get<1>(I);
-      Register FalseValue = std::get<2>(I);
-      Builder.buildInstr(MOS::G_PHI)
+    for (const auto &[Dst, TrueValue, FalseValue] : zip(Dsts, TrueValues, FalseValues)) {
+	  Builder.buildInstr(MOS::G_PHI)
```
Comment (mysterymath) [inline llvm/lib/Target/MOS/MOSLowerSelect.cpp:318]: Builder is too indented here.

**Comment by gandhi56** [comment]
Applied clang-format. 

I suppose I should have pushed my changes to a seperate commit instead of amending to the previous commit. My apologies

**Pull request merged**